### PR TITLE
Add support for windows clusters.

### PR DESCRIPTION
**Pull request by brendandburns**
@anhowe

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -399,6 +399,10 @@ def acs_create(resource_group_name, deployment_name, name, ssh_key_value, dns_na
     :param tags: Tags object.
     :type tags: object
     :param dict custom_headers: headers that will be added to the request
+    :param windows: If true, the cluster will be built for running windows container.
```
Comment (anhowe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: "Windows containers" - capital W

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -507,10 +513,19 @@ def load_acs_service_principals(config_path):
...
+                       location=None, service_principal=None, client_secret=None, windows=False, admin_password=''):
     from azure.mgmt.resource.resources.models import DeploymentProperties
     if not location:
         location = '[resourceGroup().location]'
+    windows_profile = None
+    os_type = 'Linux'
+    if windows:
+        windows_profile = {
+            "adminUsername": admin_username,
+            "adminPassword": admin_password,
```
Comment (anhowe) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Is there validation in the CLI for Windows passwords and can it be used here?  The cloud user account policy is here: https://docs.microsoft.com/en-us/azure/active-directory/active-directory-passwords-policy

Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: There's no validation in there yet (even for VMs) I'll try to add it in a different PR.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: done.

**Comment by brendandburns** [comment]
Thanks for the review. @yugang-msft can I can a review for this?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -350,7 +350,7 @@ def acs_create(resource_group_name, deployment_name, name, ssh_key_value, dns_na
                content_version=None, admin_username="azureuser", agent_count="3",
                agent_vm_size="Standard_D2_v2", location=None, master_count="3",
                orchestrator_type="dcos", service_principal=None, client_secret=None, tags=None,
-               custom_headers=None, raw=False,
+               custom_headers=None, windows=False, admin_password="", raw=False,
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Suggest update the params.py file to make `windows` as a flag, rather with values `true` and `false`

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: will the current default value of empty string work? If not,  i suggest put in a line to just throw

Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: FYI, VM's password validation is [here](https://github.com/Azure/azure-cli/blob/master/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py#L368)

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 3 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: done.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: validation added.

**Comment by brendandburns** [comment]
Comments addressed. ptal.

Thanks
--brendna

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py [hunk 4]
```diff
@@ -70,6 +70,8 @@ def _get_default_install_location(exe_name):
 register_extra_cli_argument('acs create', 'generate_ssh_keys', action='store_true', help='Generate SSH public and private key files if missing')
 register_cli_argument('acs create', 'agent_vm_size', completer=get_vm_size_completion_list)
 
+register_cli_argument('acs create', 'windows', action='store_false', help='If true, deploy a windows container cluster.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: why not `store_true`? This automatically creates a default value of False, which I believe is you want
`store_false` will default to True when the command-line argument is not present. Is that what you want?

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 5]
```diff
@@ -507,10 +513,23 @@ def load_acs_service_principals(config_path):
...
-                       location=None, service_principal=None, client_secret=None):
+                       location=None, service_principal=None, client_secret=None, windows=False, admin_password=''):
     from azure.mgmt.resource.resources.models import DeploymentProperties
     if not location:
         location = '[resourceGroup().location]'
+    windows_profile = None
+    os_type = 'Linux'
+    if windows:
+        if len(admin_password) == 0:
+            raise CLIError('--admin-password is required.')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: The upcoming check of `<6` is good enough, so I suggest we just remove this `if`

Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: I'd prefer to have a cleaner error messages.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py (hunk 4 above)
Comment (brendandburns) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: That makes no sense to me :P, but you are right I want a default value of False...

**Comment by brendandburns** [comment]
Changes addressed, please re-check.

Thanks!
--brendan

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py (hunk 4 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py]: Agreed the naming is a bit odd. It is from the built-in argparse package which CLI builds on top of

**Comment by yugangw-msft** [comment]
@brendandburns could you resolve the conflict, so i can merge?

**Comment by brendandburns** [comment]
@yugangw-msft, done.

Thanks

**Pull request merged**